src/movies.js

```diff
@@ -1,16 +1,224 @@
+    const movieTitles = copyMovies.map(elm => elm.title)
+
+    movieTitles.sort((a,b) => {
+        return a.localeCompare(b)
+    })
+
+    if (movieTitles.length > 20) {
+            movieTitles.splice(20)
+    }
+    
+    return movieTitles
+    
+}
+  
+
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(movies) {
+    const minuteMovieA = [...movies] //solo con esto no me lo coge como array distinto al original
+    
+    const minuteMovies = minuteMovieA.map (elm => { //el array era copia real pero los objetos seguían por referencia??
+        const intermediateElm = {...elm}
+        return intermediateElm
+    })
+
+    minuteMovies.forEach (elm => {
+        const hoursInMinutes = parseInt(elm.duration.substring(0, 1)) * 60
+        const minutes = parseInt(elm.duration.substring(3, 7))
+        let durationMinutes = hoursInMinutes + minutes
+        
+        if (elm.duration.includes('h') === false) {
+            durationMinutes = parseInt(elm.duration.substring(0, 3))
+        }
+
+        if (elm.duration.includes('min') === false) {
+            durationMinutes = parseInt(elm.duration.substring(0, 1) * 60)
+        }
+
+        elm.duration = parseInt(durationMinutes)
+    })
+
+    return minuteMovies
+}
+
+
 // BONUS - Iteration 8: Best yearly rate average - Best yearly rate average
+
+/* 
+    SORT movies array
+    
+    MAP contest array
+      with Object: year, rates[], average
+    
+    UNIQUIFY contest array x year
+    
+    forEach contest array(      )
+            contest.rate = movies.filter(by year)
+            contest.average = contest.rate.reduce
+    
+    SORT contest array x average
+
+    console.log
+*/
+function bestYearAvg(movies) {
+
+    if (!movies.length) {
+        return null
+    }
+    //hago copia "real" de array y elementos interiores
+    const averageMovieA = [...movies] 
+    const averageMovies = averageMovieA.map (elm => { 
+        const intermediateElm = {...elm}
+        return intermediateElm
+    })
+
+    averageMovies.sort((a,b) => a.year - b.year)
+
+    const movieYearsA = averageMovies.map(elm => elm.year) // extract the years list
+
+    const movieYears = movieYearsA.filter((elm, index) => { // and make it unique
+    return movieYearsA.indexOf(elm) == index
+    })
+
+   const contestArray = movieYears.map(elm => { //create contestArray from the years list
+       contestObject = {
+           year: elm,
+           rates: [],
+           yearAverage: 0
+       }
+       return contestObject
+   })
+
+   contestArray.forEach(elm => {
+       fillerYear = elm.year
+       elm.rates = averageMovies.filter(elm2 => elm2.year === fillerYear).map(elm => elm.rate)// add all the year's rates to the arrar
+       elm.yearAverage = elm.rates.reduce((acc,elm3) => acc + elm3, 0) / elm.rates.length// calculate the average
+   })
+
+   contestArray.sort((a,b) => b.yearAverage - a.yearAverage)
+
```

Comment:
Te falto retornar el valor (` return` `The best year was ${contestArray[0].year} with an average rate of ${contestArray[0].yearAverage}`)